staging/src/k8s.io/apimachinery/pkg/fields/selector.go

```diff
@@ -68,6 +72,13 @@ func (t *hasTerm) Matches(ls Fields) bool {
 	return ls.Get(t.field) == t.value
 }
 
+func (t *hasTerm) Supported(ls Fields) error {
+	if !ls.Has(t.field) {
```

Comment:
I would expect this to return an error and the server respond with an error - https://github.com/kubernetes/kubernetes/blob/5e5b586e55c75344ba3388c361637686526f8895/pkg/api/v1/conversion.go#L184